src/App.js

```diff
@@ -1,8 +1,112 @@
+import styled from 'styled-components';
+import { GlobalStyle } from "./GlobalStyle";
+import { useState, useCallback, useRef } from 'react';
+import InsertBox from "./component/InsertBox";
+import ListBox from "./component/ListBox";
+
+const Wrapper = styled.div`{
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: column;
+}`;
+
+const Container = styled.div`{
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  border-radius: 20px;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  width: 300px;
+  height: 700px;
+  padding: 30px;
+  justify-content: center;
+  background-color:rgb(3, 3, 36);
+  color: rgb(255, 77, 71);
+}`;
+
+const Title = styled.h1`{
+  font-size: 1.5rem;
+}`;
+
+
 function App() {
+  //toDos는 현재 들어온 할 일, setToDos는 이를 갱신하는 역할의 함수
+  const [toDos, setToDos] = useState([
+    {
+      id: 1,
+      text: '알고리즘 과제 마무리하기',
+      checked: true,
+    },
+    {
+      id: 2,
+      text: '30분 운동하기',
+      checked: false,
+    },
+  ]);
+
+  const doings = toDos.filter((todo) => !todo.checked);
+  const dones = toDos.filter((todo) => todo.checked);
+
+  //todo 추가하기
+  const nextId = useRef(3);
+  const onInsert = useCallback(
+    (text) => {
+      const todo = {
+        id: nextId.current,
+        text,
+        checked: false,
+      };
+      setToDos(toDos.concat(todo)); //toDos에 todo 추가!
+      nextId.current++; //nextId 1씩 더하기
+     },
+    [toDos]
+  );
```

Comment:
useRef를 활용하여 Insert함수를 작성하신 부분이 인상적이네요. 한번도 작성해본적 없는 형식의 코드인데 이 방법 되게 좋은것같아요!

concat함수의 연산속도가 push나 ...(spread) 연산자에 비해 많은 속도 차이가 난다고 알고 있어요. 이 부분은 아래와 같이 바꾸면 좀 더 좋을 것 같아요! [참고자료](https://www.measurethat.net/Benchmarks/Show/4223/0/array-concat-vs-spread-operator-vs-push)

```suggestion
  const nextId = useRef(3);
  const onInsert = useCallback(
    (text) => {
      const todo = {
        id: nextId.current,
        text,
        checked: false,
      };
      setToDos([...toDos, todo])
      nextId.current++; //nextId 1씩 더하기
     },
    [toDos]
  );
```

Reply:
오오 참고자료까지 너무 감사합니다!👍 속도 면에서 차이가 나는 줄은 몰랐네요!